### ISSUE TITLE
v4: Style order not working correctly with cssInterop

### ISSUE THREAD

**Issue by johnf** [marklawlor/nativewind#596]
**Describe the bug**
I don't think styles are being applied correctly at runTime when used with cssInterop

**To Reproduce**
* npx react-native@latest init NativeWindOrder
* configure for nativewind v4.alpha.30
* Use the following App.tsx

```jsx
import { cssInterop } from 'nativewind';
import React from 'react';
import { TextProps } from 'react-native';
import { Text, View } from 'react-native';
import './global.css';

// sm: 14px
// 2xl: 24px
// 5xl: 48px

/// BAD
const FontAwesome5Bad = ({
  size = 12,
  style,
  children,
}: {
  size?: number;
  style: TextProps['style'];
  children: React.ReactNode;
}) => {
  const innerStyle = [{ fontSize: size }, style];

  console.debug('Bad', innerStyle);

  return <Text style={innerStyle}>{children}</Text>;
};

cssInterop(FontAwesome5Bad, { className: { target: 'style' } });

const TextBad = ({
  className,
  children,
}: {
  className: string;
  children: React.ReactNode;
}) => <FontAwesome5Bad className={className}>{children}</FontAwesome5Bad>;

/// OK

const FontAwesome5Ok = ({
  size = 12,
  style,
  children,
}: {
  size?: number;
  style: TextProps['style'];
  children: React.ReactNode;
}) => {
  const innerStyle = [{ fontSize: size }, style];

  console.debug('OK ', innerStyle);

  return <Text style={innerStyle}>{children}</Text>;
};

const TextOk = ({
  style,
  children,
}: {
  style: TextProps['style'];
  children: React.ReactNode;
}) => <FontAwesome5Ok style={style}>{children}</FontAwesome5Ok>;

const App = () => (
  <View className="flex-1 flex-column items-center w-full mt-20">
    <TextBad className="text-5xl">SHould be large</TextBad>
    <TextOk style={{ fontSize: 48, lineHeight: 48 }}>SHould be large</TextOk>
  </View>
);
```

**Expected behavior**
Both lines should have the same font size of 48px. But the first size has 12px.

**Expo Snack**
Is there a v4 snack?

**Screenshots**
![image](https://github.com/marklawlor/nativewind/assets/42590/0ca51c94-f722-48ba-ad27-3b1aa393d747)

**Additional context**

I think the problem is here:
https://github.com/marklawlor/nativewind/blob/main/packages/react-native-css-interop/src/runtime/native/flatten-style.ts#L161-L167

```js
for (let [key, value] of Object.entries(style)) {
    if (
      // We can shortcut setting a value if it already exists
      value === undefined ||
      flatStyle[key as keyof typeof flatStyle] !== undefined
    ) {
      continue;
    }
```

Because value is already set, it skips trying to set it. I tried commenting this bit of code out, but it breaks a lot of tests in container and specificitfy.

**Comment by marklawlor** [comment]
NativeWind follows the web's style specificity - this means inline styles will override className styles (as they have a higher specificity). If you want a className style to override an inline style, you should use the [important modifier](https://tailwindcss.com/docs/configuration#important-modifier).

This is the same as NativeWind v2, you just encounter it more now that styles are passed down.

The only difference between style ordering between v2 & v4 are the ordering of className styles themselves. v4 now sorts them by specificity as well (they used to be left->right)

**Comment by marklawlor** [comment]
Also, your code does not work as you expect it to. `<FontAwesome5Bad />` will never be render with className styles because you don't pass the className down to the `<Text />`

**Comment by marklawlor** [comment]
Sorry, misunderstood the issue. I see the problem :+1:

**Comment by marklawlor** [comment]
A tentative fix has been implemented in `4.0.0-alpha.31`. I'm going to close this issue, but feel free to comment if you still have issues. This was quite a fundamental change to the library, so I wouldn't be surprised if I missed something.

Basically, I rewrote how NativeWind orders its styles. Previously it exclusively used CSS specificity, but now it uses a new ordering system that is a mix of both React Native and CSS specificity. I've hastily documented my thoughts about it here https://github.com/marklawlor/nativewind/blob/main/apps/website/docs/core-concepts/style-specificity.mdx. Ignore the bad english, this  was basically a thought dump, but hopefully it helps explains why the solution is a bit complex.

**Comment by johnf** [comment]
@marklawlor I've done some initial testing and this looks good so far. I'll raise another issue if I notice any issues.

Also, thanks for the amazing turn around on all these issues - especially the docs on this one. The complexity here is quite interesting.
